iptables/utils_for_test.go

```diff
@@ -245,6 +246,15 @@ func (d *restoreCmd) Run() error {
 		log.Warn("Simulating an iptables-restore failure")
 		return errors.New("Simulated failure")
 	}
+	// Alter the dataplane before the restore is applied.
+	if d.Dataplane.InsertRandomRuleInForward {
```

Comment:
Just spotted that we already had a hook `OnPreRestore` that lets you attach a function to call before the restore happens.  Not sure it's worth switching to that mechanism since this is nice and simple but always worth looking around the code to see if there's already a way to get what you need.

Reply:
Nice, I was looking for something like that but missed it.